packages/odyssey-react/src/components/Button/Button.tsx

```diff
@@ -75,12 +76,44 @@ let Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
     wide && styles.wideLayout
   );
 
+  let textColor: TextProps["color"] = "bodyInverse";
+  let hoverColor: TextProps["hoverColor"] = "bodyInverse";
+  let focusColor: TextProps["focusColor"] = "bodyInverse";
+  let disabledColor: TextProps["disabledColor"] = "bodyInverse";
+
+  if (variant === "secondary" || variant === "clear") {
+    textColor = "primary";
+    focusColor = "primary";
+    disabledColor = "primaryLight";
+  }
+
+  if (variant === "clear") {
+    hoverColor = "primaryDark";
+  }
+
+  if (variant === "dismiss") {
+    textColor = "body";
+    hoverColor = "body";
+    focusColor = "body";
+    disabledColor = "sub";
+  }
```

Comment:
this works - but its also screaming out for a simple pure function utility well named like `buildColors` or similar. That would help to cleanup the component function body here.

```
const { textColor, focusColor, disabledColor, hoverColor } = buildColors(variant)
```